fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -279,7 +316,8 @@ public void writeData(long id, int block, @Nonnull byte[] value) {
     @Override
     public void deleteFile(@Nonnull String name) throws IOException {
         LOGGER.trace("deleteFile -> {}", name);
-        boolean deleted = getFDBLuceneFileReference(name).thenApplyAsync(
+        long start = System.nanoTime();
+        boolean deleted = context.instrument(FDBStoreTimer.Events.LUCENE_DELETE_FILE,getFDBLuceneFileReference(name).thenApplyAsync(
```

Comment:
Similar to writes, deletes are cached entirely in the client until commit, so I think the only thing you'll really be timing here is `getFDBLuceneFileReference()`.

Reply:
I know this isn't part of this PR, but you don't need (or want) to use `thenApplyAsync()`.  Most `*Async()` calls like this, will create a new future that will execute the code in the lambda -- so the future is created then enqueued for execution in the `Executor` threads.  This is typically only done if the lambda is costly and you don't want this chain of futures to wait for its completion.  In this case (and, by far our most common cases), the cost of the labda execution is minimal (since `clear()` doesn't involve any actual I/O -- and, if it did, it would have returned a future), so it is way more costly do have `thenApplyAsync()` dispatch a new work item, than to just use `thenApply()`

Reply:
I know this is super pendantic, but please try to put spaces after commas, like:

    instrument(FDBStoreTimer.Events.LUCENE_DELETE_FILE, getFDBLuceneFileReference(name)

the main thing (for me) is when reviewing, without the space, it looks like a single expression.

Reply:
I _think_ I got all of these, but since they mostly just go completely through my brain without my being able to tell them, I may have missed a couple--my preemptive apologies :)